Makefile

```diff
@@ -239,45 +307,9 @@ $(BINDIR)/.:
 $(BINDIR)/%/.:
 	$(COMMAND) mkdir -p $@
 
-# ###################################################################
-# The dependencies for each test-use-cases/ sample program.
-# Every example program of the form bin/test-use-cases/<eg-prog> depends on
-# obj/test-use-cases/<eg-prog>/*.o -> *.c
-# There can be more than one .o's linked to create test-use-cases example
-# program.
-
-# ----
-SINGLE_FILE_C_PROGRAM_TESTSRC := $(shell find $(TESTSDIR)/single-file-C-program -type f -name *.c -print)
-
-SINGLE_FILE_C_PROGRAM_TESTSRC += $(L3_SRCS)
-
-SINGLE_FILE_C_PROGRAM_OBJS := $(SINGLE_FILE_C_PROGRAM_TESTSRC:%.c=$(OBJDIR)/%.o)
-
-$(BINDIR)/$(TESTSDIR)/single-file-C-program: $(SINGLE_FILE_C_PROGRAM_OBJS)
-
-# ----
-SINGLE_FILE_CPP_PROGRAM_TESTSRC := $(shell find $(TESTSDIR)/single-file-Cpp-program -type f -name *.cpp -print)
-SINGLE_FILE_CPP_PROGRAM_TESTSRC += $(L3_SRCS)
-
-# For C++ sources, have to go back and replace l3.c -> l3.o
-SINGLE_FILE_CPP_PROGRAM_OBJS := $(SINGLE_FILE_CPP_PROGRAM_TESTSRC:%.cpp=$(OBJDIR)/%.o)
-SINGLE_FILE_CPP_PROGRAM_OBJS := $(SINGLE_FILE_CPP_PROGRAM_OBJS:%.c=$(OBJDIR)/%.o)
-
-$(BINDIR)/$(TESTSDIR)/single-file-Cpp-program: $(SINGLE_FILE_CPP_PROGRAM_OBJS)
-
-# ----
-SINGLE_FILE_CC_PROGRAM_TESTSRC := $(shell find $(TESTSDIR)/single-file-CC-program -type f -name *.cc -print)
-SINGLE_FILE_CC_PROGRAM_TESTSRC += $(L3_SRCS)
-
-# For C++ sources, have to go back and replace l3.c -> l3.o
-SINGLE_FILE_CC_PROGRAM_OBJS := $(SINGLE_FILE_CC_PROGRAM_TESTSRC:%.cc=$(OBJDIR)/%.o)
-SINGLE_FILE_CC_PROGRAM_OBJS := $(SINGLE_FILE_CC_PROGRAM_OBJS:%.c=$(OBJDIR)/%.o)
-
-$(BINDIR)/$(TESTSDIR)/single-file-CC-program: $(SINGLE_FILE_CC_PROGRAM_OBJS)
-
```

Comment:
This whole deleted block is now unnecessary as the dependencies of make targets on sample program binaries is defined differently, above.